code/game/gamemodes/objectives/_objective.dm

```diff
@@ -225,13 +225,26 @@ GLOBAL_LIST(admin_objective_list) //Prefilled admin assignable objective list
 		explanation_text = team_explanation_text
 
 /datum/objective/proc/generate_stash(list/special_equipment)
-	var/datum/mind/receiver = pick(get_owners())
-	var/obj/item/storage/backpack/satchel/flat/empty/secret_bag = receiver.antag_stash
+	var/list/owners = get_owners()
+	var/datum/mind/tester = pick(owners)
+	var/obj/item/storage/secret_bag = null
+	for (var/datum/component/stash/stash in tester.antag_stashes)
+		// Is this a valid stash
+		// Must be owned exclusively by our owners
+		var/valid = TRUE
+		for (var/datum/mind/mind in stash.stash_minds)
+			if (!(mind in team.members))
+				valid = FALSE
+				break
```

Comment:
this can be an issue when an antag has more than single antag role (e.g. traitor + BB + incursion), and received more stashes. stash mind will be single, but it qualifies a massive team stash because their mind is in team. This means a single person directive can go into a team stash.
I think there should be a team proc code to get their team stash?

Reply:
Good spot, a single person directive cannot go into a team stash, however a team item can go into a single person's directive.
For an existing stash to be used, there must be no members that aren't in the team present in the stash, but it doesn't check that all members of the team own that stash.